### PR TITLE
Calibration: Implement temperature lookup-table based calibration

### PR DESCRIPTION
**Pull request by adisuciu**
Signed-off-by: Adrian Suciu <adrian.suciu@analog.com>

### CODE REVIEW

File: src/calibration.cpp [hunk 1]
```diff
@@ -135,6 +135,16 @@ void Calibration::cancelCalibration()
 	m_cancel = true;
 }
 
+bool Calibration::hasContextCalibration()
```
Comment (DanielGuramulta) [inline src/calibration.cpp:138]: Can this method be const? Does it modify the state of the Calibration object?

File: src/info_page.cpp [hunk 2]
```diff
@@ -170,8 +170,10 @@ QPair<bool, QString> InfoPage::translateInfoParams(QString key)
 	} else if (key.contains("ip")) {
 		key = "IP Address";
 	} else if (key.startsWith("cal,gain") ||
-		   key.startsWith("cal,offset")) {
-		key = "";
+		   key.startsWith("cal,offset") ||
+		   key.startsWith("cal,temp_lut")) {
+//		key = "";
```
Comment (DanielGuramulta) [inline src/info_page.cpp:175]: nitpick: if this is not needed it should be removed

File: src/info_page.cpp [hunk 3]
```diff
@@ -401,6 +407,12 @@ M2kInfoPage::M2kInfoPage(QString uri,
 	ui->extraWidget->setMinimumHeight(700);
 	setConnectionStatusLabel(tr("Not connected"));
 	setCalibrationInfoLabel(tr("Always disconnect analog inputs/outputs before calibration"));
+	if(ctx)
```
Comment (DanielGuramulta) [inline src/info_page.cpp:410]:
```cpp
if (..) {
    ...
}
```

File: src/info_page.cpp [hunk 4]
```diff
@@ -412,7 +424,43 @@ M2kInfoPage::~M2kInfoPage()
...
 {
 	InfoPage::getDeviceInfo();
-	//refreshTemperature();
+	refreshTemperature();
+}
+
+void M2kInfoPage::refreshTemperature()
+{
+
+	libm2k::context::M2k *temp_m2k;
```
Comment (DanielGuramulta) [inline src/info_page.cpp:433]: Initialise pointer to `nullptr`

File: src/info_page.cpp [hunk 5]
```diff
@@ -412,7 +424,43 @@ M2kInfoPage::~M2kInfoPage()
...
+	if(!m_ctx)
```
Comment (DanielGuramulta) [inline src/info_page.cpp:434]:
```cpp
if () {
  ...
} else {
  ...
}

File: src/info_page.cpp [hunk 6]
```diff
@@ -412,7 +424,43 @@ M2kInfoPage::~M2kInfoPage()
...
+	{
+		temp_m2k = libm2k::context::m2kOpen(m_uri.toLocal8Bit().constData());
+	}
+	else
+		temp_m2k = m_m2k;
+
+	if(temp_m2k)
```
Comment (DanielGuramulta) [inline src/info_page.cpp:441]:
```cpp
if (..) {
    ...
}
```

File: src/info_page.cpp [hunk 7]
```diff
@@ -412,7 +424,43 @@ M2kInfoPage::~M2kInfoPage()
...
+	{
+		auto dmm = temp_m2k->getDMM("ad9963");
+		auto ch = dmm->readChannel("temp0");
+		auto val = ch.value;
+		m_info_params["Temperature"] = QString::number(val);
+	}
+	if(!m_ctx)
```
Comment (DanielGuramulta) [inline src/info_page.cpp:448]:
```cpp
if (..) {
    ...
}
```

File: src/info_page.cpp [hunk 8]
```diff
@@ -412,7 +424,43 @@ M2kInfoPage::~M2kInfoPage()
...
+	{
+		libm2k::context::contextClose(temp_m2k, false);
+	}
+	qDebug()<<"temperature is" <<m_info_params["Temperature"];
```
Comment (DanielGuramulta) [inline src/info_page.cpp:452]: Nitpick: missing spaces `<< "tem..." << m_info....`

File: src/info_page.hpp [hunk 9]
```diff
@@ -118,16 +122,22 @@ class M2kInfoPage : public InfoPage
 			     QWidget *parent = 0);
 	~M2kInfoPage();
 	void getDeviceInfo();
+	void setCtx(iio_context *ctx);
```
Comment (DanielGuramulta) [inline src/info_page.hpp:125]: missing `override` specifier?

File: src/calibration.cpp (hunk 1 above)
Comment (adisuciu) [reply src/calibration.cpp]: fixed

File: src/info_page.cpp (hunk 2 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/info_page.cpp (hunk 3 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/info_page.cpp (hunk 4 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/info_page.cpp (hunk 6 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/info_page.cpp (hunk 8 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/info_page.hpp (hunk 9 above)
Comment (adisuciu) [reply src/info_page.hpp]: fixed

File: src/info_page.cpp (hunk 7 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/info_page.cpp (hunk 5 above)
Comment (adisuciu) [reply src/info_page.cpp]: fixed

File: src/calibration.cpp [hunk 10]
```diff
@@ -135,6 +135,16 @@ void Calibration::cancelCalibration()
 	m_cancel = true;
 }
 
+const bool Calibration::hasContextCalibration()
```
Comment (DanielGuramulta) [inline src/calibration.cpp:138]: I was actually referring to the method signature here `bool Calibration::hasContextCalibration() const`

**Review by DanielGuramulta — APPROVED** [review]
Thank you for addressing the issues, other than the const correctness remark rest looks good!

**Pull request merged**